logstash-core/lib/logstash/api/rack_app.rb

```diff
@@ -8,9 +8,84 @@
 module LogStash
   module Api
     module RackApp
-      def self.app
+      # Cabin is not compatible with CommonLogger, and this gives us more control anyway
+      class ApiLogger
```

Comment:
I understand the motivations your are going with this solution, and not saying is bad, just a few open questions I have that would be ncie to provide closure somehow.

* What do you think about performance? as more rack middleware we use, more roundtrips the request is doing, have you thought about it? 

* This Rack applications should be properly tested as they include another layer of interaction for every request that hits the api.